### ISSUE TITLE
Incomplete aria-expanded in navigation menu

### ISSUE THREAD

**Issue by metamoni** [joemasilotti/railsdevs.com#540]
## Describe the bug
The guest/user navigation menu has `aria-expanded="false"`, but this doesn't update to `true` when the menu is expanded. This affects both desktop and mobile views.

Views:
`app/components/nav_bar/user_component.html.erb`
`app/components/nav_bar/user_component.html.erb`

## To reproduce

Steps to reproduce the behavior (on macOS):

1. Turn on VoiceOver: `command + F5`
2. Tab to your avatar in the top navigation bar and notice the screen reader announces the menu is collapsed
3. Hit "Enter" to expand the menu and notice the screen reader announces "menu (3 items)
4. Turn VoiceOver off: `command + F5`

## Expected behavior
When the menu is open, the screen reader should announce that the menu is expanded
When the menu is closed again, the screen reader should announce "collapsed"

## Actual behavior
Screen reader does not announce "expanded" when menu is open
Screen reader does not announce "collapsed" when menu is closed after an open state

## Screenshots
_desktop view_
https://user-images.githubusercontent.com/22390758/177098093-a254b34e-0856-4a86-91fa-990b15026bab.mov

_mobile view_

https://user-images.githubusercontent.com/22390758/177098529-4722a4ac-fcbf-4414-af0c-3c28137a23e7.mov

## Additional context
Correct usage of `aria-expanded` (example): https://w3c.github.io/aria-practices/examples/accordion/accordion.html

**Comment by axhgl** [comment]
Since I've already dealt with a similar one (#539), should I give this one a try, as well?

**Comment by joemasilotti** [comment]
Closed via #585.
